markets/perps-market/contracts/storage/AsyncOrder.sol

```diff
@@ -348,11 +350,28 @@ library AsyncOrder {
             revert InsufficientMargin(runtime.currentAvailableMargin, runtime.totalRequiredMargin);
         }
 
-        int256 lockedCreditDelta = perpsMarketData.requiredCreditForSize(
-            MathUtil.abs(runtime.newPositionSize).toInt() - MathUtil.abs(oldPosition.size).toInt(),
-            PerpsPrice.Tolerance.DEFAULT
-        );
-        GlobalPerpsMarket.load().validateMarketCapacity(lockedCreditDelta);
+        /// @dev if new position size is not 0, further credit validation required
+        if (runtime.newPositionSize != 0) {
+            /// @custom:magnitude determines if more market credit is required
+            /// when a position's magnitude is increased, more credit is required and risk increases
+            /// when a position's magnitude is decreased, less credit is required and risk decreases
+            uint newMagnitude = MathUtil.abs(runtime.newPositionSize);
+            uint oldMagnitude = MathUtil.abs(oldPosition.size);
+
+            /// @custom:side reflects if position is long or short; if side changes, further validation required
+            /// given new position size cannot be zero, it is inconsequential if old size is zero; 
+            /// magnitude will necessarily be larger
+            bool sameSide = runtime.newPositionSize > 0 == oldPosition.size > 0;
```

Comment:
if `oldPosition.size` is 100 eth and `runtime.newPositionSize` is zero, this result in False, right? While it should True?

Reply:
you are correct. however, given previous logic, `runtime.newPositionSize` will never be zero.

i.e.,

```
/// @dev if new position size is not 0, further credit validation required
if (runtime.newPositionSize != 0) {
    (...)
}
```

I also made not of this in the comment above the `sameSide` assignment: "given new position size cannot be zero"

Reply:
np. If I'm not wrong, there's a sameSide helper that can be used here to prevent duplicated code (though is not A LOT of code)